### PR TITLE
Support unenclosed inner text for details elements in to be visible

### PR DESCRIPTION
**Pull request by kschow**
**What**:

Supporting unenclosed inner text inside ```<details />``` elements for toBeVisible(). See https://github.com/testing-library/jest-dom/issues/395

**Why**:

Unenclosed inner text is valid for ```<details />``` elements and should therefore be supported.

**How**:

Updated the isAttributeVisible function. In the case where the inner text is unenclosed there is no "previousElement" so defaults to only check the open attribute of the details element itself. Otherwise, it follows the same logic as before.

**Checklist**:

- [x] Documentation
- [x] Tests
- [ ] Updated Type Definitions "N/A"
- [x] Ready to be merged

### CODE REVIEW

File: README.md
Comment (kschow) [inline README.md:52]: I think this, and the other formatting change might have been automatic, I didn't do them intentionally but I can try to revert those changes if necessary.

File: src/__tests__/to-be-visible.js
Comment (kschow) [inline src/__tests__/to-be-visible.js:320]: This set of tests might be a bit redundant, I added them because I wasn't sure how the unenclosed and enclosed details would interact. But thinking it over, there really shouldn't be a significant difference as opposed to the previous nested tests (which these are based off).

**Comment by feedmypixel** [comment]
I've just come across this issue when testing details with `"@testing-library/jest-dom": "5.16.2"`. This PR looks a little stale? any help needed?

**Comment by kschow** [comment]
I don't think I need any help with the actual code. There were a couple
lint issues that I ended up fixing and I don't think it re-ran the checks
to progress the status of the issue.

Feel free to do what you think is best here. It'd probably need to be
rebased or have the current master merged in because I'm sure things have
changed.

**Comment by feedmypixel** [comment]
> Feel free to do what you think is best here. 

I have rebased main. All linting and testing is passing, can you add me as a collaborator on https://github.com/kschow/jest-dom/ and I will raise a PR to merge into your branch. Once you have merged that into yours that then should kick the status required checks here and we will be moving forward

**Comment by feedmypixel** [comment]
Ok thats been raised on your repo and merged into your branch - i'll leave it up to you but thats your original PR with an updated main - its also kicked off the status checks here

**Comment by kschow** [comment]
I think at this point we're waiting on a maintainer to take a look at this and approve it.

**Review by gnapse — APPROVED** [review]

**Pull request merged**
